File: workflow-hello/src/main/java/com/acme/workflows/hello/java/HelloWorkflowImpl.java
package com.acme.workflows.hello.java;

import com.acme.contracts.services.hello.HelloService;
import com.acme.common.AbstractWorkflow;
import com.acme.contracts.workflows.hello.HelloWorkflow;

public class HelloWorkflowImpl extends AbstractWorkflow implements HelloWorkflow {
    private final HelloService helloService = newService(HelloService.class);

    @Override public String greet(String name) {
        String str = helloService.sayHello(name);

        String greeting = helloService.addEnthusiasm(str);

        log(greeting);

        return greeting;
    }
}

File: workflow-hello/src/main/kotlin/com/acme/workflows/hello/kotlin/HelloWorkflowImpl.kt
package com.acme.workflows.hello.kotlin

import com.acme.contracts.services.hello.HelloService
import com.acme.common.AbstractWorkflow
import com.acme.contracts.workflows.hello.HelloWorkflow

class HelloWorkflowImpl : AbstractWorkflow(), HelloWorkflow {

    private val helloService = newService(HelloService::class.java)

    override fun greet(name: String): String {
        val str = helloService.sayHello(name)

        val greeting = helloService.addEnthusiasm(str)

        log(greeting)

        return greeting
    }
}
